### PR TITLE
fixed bug in int(). Added char(), unchar(), byte(), boolean(), str(), he...

### PR DESCRIPTION
**Pull request by russomf**
...x() and unhex()

I have given char() and other conversion functions a lot of consideration. My proposal is described below. I will follow up with a pull request of all my proposed additions.

First I should say that I found and fixed a serious bug in p5.prototype.int() that occurs when an array of values is provided as the parameter. For example, int([1, 2, "100"]) should produce [1, 2, 100], but it is currently producing [1, 2, 4]). After some investigation I found that the problem has to do with incorrect usage of Array.prototype.map(), which invokes the given function on all members of Array. In fact, the function parameter to map() is passed three values: (1) Array element, (2) Array index and (3) Array. The function p5.prototype.int() takes two parameters: (1) value and (2) radix. When an Array is passed as a parameter to p5.prototype.int() and Array.prototype.map() is invoked on each element, the array index is passed to p5.prototype.int() as the second parameter and interpreted as the radix, which produces some strange results. To fix the problem I wrapped p5.prototype.int() in an anonymous function with radix set and passed this to map(). It works correctly now.

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

As far as the other conversion functions go, I decided to let what I believe to be the most common use cases inform the design decisions. 

__char()__

I believe that the typical user will expect the char() conversion function to take an integer and return a character. But because JavaScript has no char data type, it must return a single-character string. If a string is passed, char() tries to parse it as an integer first. If an array is passed, char() is applied recursively to all elements of the array (like int()). For the sake of performance, an iterative implementation may be a better choice at some point. At its core, char() makes use of JavaScript's String.fromCharCode(n).

This design decision left a pretty big hole in P5 - that is, with the new char() conversion function there is now a way to turn an integer into a single-character string, but there is no way to turn a single-character string back into an integer. This capability would correspond to JavaScript's n.charCodeAt(0) function. In Processing (and ProcessingJS) the byte() function serves this purpose, and that makes sense because both have the notion of a char data type. For example, in Processing, byte('1') returns 49, because '1' is a char with the ASCII value of 49. In JavaScript, strings that resemble numbers sometimes freely convert to numbers. For example, the expression +'1' results in the number 1, not an error. For this reason I thought it was important that P5 have an explicit function that turns a single-digit string to a number (it's char code). In the spirit of Processing's unhex() and unbinary() functions, I am proposing the introduction of unchar().

__unchar()__

The new unchar() function takes a single-character string and returns the character code of that character. It also takes an array, as do all the new conversion functions, to be described. With unchar(), there is no ambiguity. For example, unchar("9") returns 57, the character code for "9" and byte("9") returns 9, the conversion of the string representation "9" into a number. (I also added the byte() conversion function and others.)

__char() and unchar() with split() and join()__

With the support of arrays in both the char() and unchar() conversion functions, it is possible to combine with split() and join() to make a nice round trip possible.

```javascript
var strng  = "ABC";
var chars  = split(strng, "");   // ["A", "B", "C"]
var nums   = unchar(chars);      // [65, 66, 67]
var chars2 = char(nums);         // ["A", "B", "C"]
var strng2 = join(chars2, "");   // "ABC"
```

As mentioned, while it was at it I also implemented several other conversion functions, including:  byte(), boolean(), str(), hex(), and unhex(). Please see the pull request for details.

### CODE REVIEW

**Comment by russomf** [comment]
I'm curious to know if there are comments or questions regarding this pull request.

I should add that hex() and unhex() are heavily inspired by the implementation in processingjs.

File: src/data/conversion.js [hunk 1]
```diff
@@ -57,11 +57,213 @@ define(function (require) {
...
+   * print(boolean([0, 12, "true"])); // [false, true, false]
+   * </code></div>
+   */
+  p5.prototype.boolean = function(n) {
+    if (typeof n === 'number') {
+      if (n === 0) {
+        return false;
+      } else {
+        return true;
+      }
```
Comment (lmccart) [inline src/data/conversion.js]: I think we could condense this and replace lines 112-116 with:
return n !== 0;

File: src/data/conversion.js [hunk 2]
```diff
@@ -57,11 +57,213 @@ define(function (require) {
...
+    } else if (typeof n === 'string') {
+      if (n.toLowerCase() === 'true') {
+        return true;
+      } else {
+        return false;
+      }
```
Comment (lmccart) [inline src/data/conversion.js]: Same thing here: 
return n.toLowerCase() === 'true';

**Comment by lmccart** [comment]
Thanks @russomf this looks great! Apologies for the delay, I was traveling this past week. Do you mind making the couple small changes I noted, and then I can merge it in?

**Comment by russomf** [comment]
Great idea. Changes made and pushed.

**Comment by lmccart** [comment]
wonderful, thank you!

**Pull request merged**